samza-test/src/test/java/org/apache/samza/test/framework/StreamTaskIntegrationTest.java

```diff
@@ -123,7 +124,7 @@ public void testSyncTaskWithMultiplePartition() throws Exception {
         .getOutputDescriptor("output", new NoOpSerde<Integer>());
 
     TestRunner
-        .of(MyStreamTestTask.class)
+        .of(new LegacyTaskApplication(MyStreamTestTask.class.getName()))
```

Comment:
FWIW, please take a look at PR #642 , most of the changes here are already incorporated in that PR. Question here: any specific reason that you want to explicitly construct a LegacyTaskApplication here, instead of retaining the same API, but internally convert the MyStreamTestTask.class to a LegacyTaskApplication instance in TestRunner?

Reply:
We have three choices for Public Api

Option 1: `TestRunner.of(SamzaApplication)`, `TestRunner.of(AsyncStreamTask)`, `TestRunner.of(StreamTask)`

Option 2:  `TestRunner.of(SamzaApplication)`, `TestRunner.of(LegacyTaskApplication)`,

Option 3: `TestRunner.of(SamzaApplication)`, `TestRunner.of(Class taskClass)`

I prefer the option 2 makes it more straighforward for end user, option 3 is very generic and vague (which I want to remove)

I am ok with either Option 1 or Option 2. I will leave it up to the reviewers to suggest

Note: when i was read Samza-1818, i thought I am responsible for the refacotr, if #642 gets merged first I can rebase the changes

Reply:
What's the specific concern on option 3? Type safety? We can add that in the TestRunner.of() method as well. If your concern is that what end user will see as a supported implementation of application, we already support the legacy case that end user only implements the task class anyways and I don't think that will add additional confusion. In fact, exposing the constructor of LegacyTaskApplication is worse, since we don't expect users construct the LegacyTaskApplication at all in the supported APIs.

Reply:
Sure, will revert the changes